rmgpy/rmg/input.py

```diff
@@ -174,25 +174,50 @@ def convert_binding_energies(binding_energies):
     return new_dict
 
 
-def species(label, structure, reactive=True):
+def species(label, structure, reactive=True, cut=False, size_threshold=None):
     logging.debug('Found {0} species "{1}" ({2})'.format('reactive' if reactive else 'nonreactive',
                                                          label,
                                                          structure.to_smiles()))
 
     if '+' in label:
         raise InputError('species {0} label cannot include a + sign'.format(label))
 
-    try:
-        spec, is_new = rmg.reaction_model.make_new_species(structure, label=label, reactive=reactive)
-    except:
-        logging.error(f'Error when reading species "{label}" from input file.')
-        raise
-    if not is_new:
-        raise InputError("Species {0} is a duplicate of {1}. Species in input file must be unique".format(label,
-                                                                                                          spec.label))
+    if cut:
+        mol_to_frag[label] = {} # key:original molecule label, value:created fragment label
+        if size_threshold:
+            cut_frag_list = structure.cut_molecule(size_threshold=size_threshold)
+        else:
+            cut_frag_list = structure.cut_molecule()
+        logging.info('The original molecule {0} is divided into several fragments:'.format(label))
+        for initial_frag in cut_frag_list:
+            frag_label = initial_frag.to_smiles()
+
+            logging.info(frag_label)
+
+            if frag_label not in mol_to_frag[label]:
+                mol_to_frag[label][frag_label] = 1
+            else:
+                mol_to_frag[label][frag_label] += 1
+
+            spec, is_new = rmg.reaction_model.make_new_species(initial_frag, label=frag_label, reactive=reactive)
+            # if duplicated fragment is created, combine and add to initialMoleFractions
+            if not is_new:
+                continue
+
+            rmg.initial_species.append(spec)
+            species_dict[frag_label] = spec
```

Comment:
This one is in the for loop which checks whether `spec` is new or not. If not `is_new`, it will not be added into `rmg.initial_species` and `species_dict`.